flink-runtime/src/test/java/org/apache/flink/runtime/execution/librarycache/BlobLibraryCacheManagerTest.java

```diff
@@ -116,21 +119,21 @@ public void testLibraryCacheManagerJobCleanup() throws IOException, InterruptedE
 			checkFileCountForJob(1, jobId2, server);
 			checkFileCountForJob(0, jobId2, cache);
 
-			libCache.registerJob(jobId2, keys2, Collections.<URL>emptyList());
-			ClassLoader classLoader2 = libCache.getClassLoader(jobId2);
+			final LibraryCacheManager.ClassLoaderLease classLoaderLeaseJob2 = libCache.registerClassLoaderLease(jobId2);
+			final ClassLoader classLoader2 = classLoaderLeaseJob2.getOrResolveClassLoader(keys2, Collections.emptyList());
 			assertNotEquals(classLoader1, classLoader2);
 
 			try {
-				libCache.registerJob(jobId2, keys1, Collections.<URL>emptyList());
+				classLoaderLeaseJob2.getOrResolveClassLoader(keys1, Collections.<URL>emptyList());
 				fail("Should fail with an IllegalStateException");
 			}
 			catch (IllegalStateException e) {
 				// that's what we want
 			}
 
 			try {
-				libCache.registerJob(
-					jobId2, keys2,
+				classLoaderLeaseJob1.getOrResolveClassLoader(
```

Comment:
are we intentionally using the least for job 1 with the keys of job 2?

Reply:
No this is wrong. It should be `classLoaderLeaseJob2` here. I will correct it.